src/components/GraphFilter/GraphRefresh.tsx

```diff
@@ -73,20 +71,16 @@ export class GraphRefresh extends React.PureComponent<GraphRefreshProps> {
   }
 
   render() {
-    const retention = this.props.serverConfig.prometheus.storageTsdbRetention;
-    const validDurations = getValidDurations(GraphRefresh.DURATION_LIST, retention);
-    const validDuration = getValidDuration(validDurations, this.props.duration);
-
     return (
       <>
         <label className={GraphRefresh.durationLabelStyle}>Fetching</label>
         <ToolbarDropdown
           id={'graph_filter_duration'}
           disabled={this.props.disabled}
-          handleSelect={this.handleDurationChange}
-          value={validDuration}
-          label={String(validDurations[validDuration])}
-          options={validDurations}
+          handleSelect={key => this.props.setDuration(Number(key))}
+          value={this.props.duration}
```

Comment:
I see, ok, I'll put it back

Reply:
Looking at that more closely, I'm not sure it's really recommended to change duration in "render" like that. It would mean that, if I set an incorrect value from the URL, the dropdown will show something different that what is actually used to perform the prometheus queries. Which could really be misleading.

Or we would have to make a broader refactor to validate durations not on rendering, but on URL parsing.

Reply:
Yes, you are right in that it's better a layer of URL validation. I think that's under the scope of this Jira: https://issues.jboss.org/browse/KIALI-420

In the meantime, I think it's not bad to do that here.

Reply:
Ok, so moving this validation to URL parsing : see commit 914d7a2